### ISSUE TITLE
InputEvents not binding correctly

### ISSUE THREAD

**Issue by ahumphreys87** [akre54/Backbone.NativeView#26]
Hi, 

I have the following events hash:

```
events: {
  'blur input': 'someHandler
}
```

`someHandler` is never called> I _think_ this may be because `NativeView` is relying on the Event propagating up the tree and checking whether selectors match on the eventTarget. Is the issue that InputEvents such as `blur`, `input` or `change` don't get propagated up to non-input nodes?

Would a better way to attach events be to query the view for the selector and use `addEventListener` on the Elements that returns rather than the `View`s root element?

**Comment by ahumphreys87** [comment]
Noticed this works if we change: https://github.com/akre54/Backbone.NativeView/blob/master/backbone.nativeview.js#L125 to enable `useCapture`

`elementAddEventListener.call(this.el, eventName, handler, true);`

Any objections to a PR for this?

**Comment by akre54** [comment]
This would be a breaking change. Do you see any downsides?

**Comment by akre54** [comment]
We could also specifically enable `useCapture` only for `change` and `submit` events which might solve this.

**Comment by ahumphreys87** [comment]
I haven't noticed any but haven't done any perf tests. Maybe performance would be effected slightly?

Not sure if it would be breaking. The API would still work as it is to the user, would just fix this bug

**Comment by ahumphreys87** [comment]
What about other input specific events? Blur and input seem to be effected also

**Comment by akre54** [comment]
The problem with `useCapture` is that it would break users' existing `stopPropagation` calls. The events would be out of order which is potentially a big deal. We definitely can't enable it by default.

Is it obvious how jQuery does it?

**Comment by ahumphreys87** [comment]
Hmm let me investigate that and I'll post back :)

Thanks btw, I'm build a Marionette.Native plugin. This library and
Backbone.Fetch are the main parts of it, nice work!

**Comment by ahumphreys87** [comment]
So looking at the jquery source, they seem to attach the events to each element individually rather than the document: https://github.com/jquery/jquery/blob/76c5237cc408c01ae67f5c68d39ed6f1e4ef8003/src/event.js#L92-L94

This would probably be quite a large piece of work for this library as we'd need to manage unbinding as well. What do you think @akre54 ?

**Comment by akre54** [comment]
Right. (sorry I was thinking of jQuery's [`delegate`](http://api.jquery.com/delegate/)*, or the React event system which *does* handle event bubbling). This may be a 'known issue' if we can't work around it without writing a huge layer on top of native events. Any ideas?

* Also the 4-arg form of `$.fn.on`

**Comment by akre54** [comment]
I'm split on this. I think our two options are to whitelist events that don't bubble (`blur`, `change`, etc) and special-case them with the `useCapture` flag, or acknowledge that these are special events and require you to bind the event handler yourself:

```js
_.first(this.$('input')).addEventListener('change', () => {})
```

Both have their downsides, but also aren't horrific in the grand scheme of things.

**Comment by ahumphreys87** [comment]
Yeah I agree, the `$` way of searching for elems with the selector and attaching events to those elems rather than document seems like overkill here.

I'd be happy with a whitelist - I'm yet to see adverse effects using `useCapture` so this may well be a good compromise.

**Comment by akre54** [comment]
Sure thing. Wanna open a pull with an implementation?

>

**Comment by ahumphreys87** [comment]
Yeah sure, ill get on it tomorrow morning (UK based 😄)

**Comment by cloakedninjas** [comment]
Seen this?

https://github.com/samriley/Backbone.NativeView/commit/c0d8210471923243f19ac288fbc10bd354ae98de
